graphs/possible_bipartition.py

```diff
@@ -5,8 +5,48 @@ def possible_bipartition(dislikes):
     """ Will return True or False if the given graph
         can be bipartitioned without neighboring nodes put
         into the same partition.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    
+    if not dislikes:
+        return True
 
+    deck = deque()
+    visited = set()
+    left = set()
+    right = set()
+
+    if dislikes[0]:
+        start = 0
+    else:
+        start = 1
```

Comment:
Imagine a graph that has multiple disconnected nodes stored at the beginning of the adjacency list `dislikes`. How might you refactor your code to account for this edge case?